spec/lib/two_factor_authentication/models/two_factor_authenticatable_spec.rb

```diff
@@ -2,154 +2,172 @@
 include AuthenticatedModelHelper
 
 describe Devise::Models::TwoFactorAuthenticatable do
-  describe '#otp_code' do
-    shared_examples 'otp_code' do |instance|
-      subject { instance.otp_code(time) }
-      let(:time) { 1_392_852_456 }
+  describe '#create_direct_otp' do
+    let(:instance) { build_guest_user }
 
-      it 'returns an error if no secret is set' do
-        expect { subject }.to raise_error Exception
-      end
+    it 'set direct_otp field' do
+      expect(instance.direct_otp).to be_nil
+      instance.create_direct_otp
+      expect(instance.direct_otp).not_to be_nil
+    end
 
-      context 'secret is set' do
-        before :each do
-          instance.otp_secret_key = '2z6hxkdwi3uvrnpn'
-        end
+    it 'set direct_otp_send_at field to current time' do
+      Timecop.freeze()
```

Comment:
When freezing time outside of a block, `Timecop.return` must always be called at the end of the test to avoid any time-based issues with subsequent tests. Alternatively, always use a block with Timecop.freeze, such as:

```ruby
Timecop.freeze(Time.now) do
  instance.create_direct_otp
  expect(instance.direct_otp_sent_at).to eq(Time.now)
end
```
See https://github.com/travisjeffery/timecop#timecopsafe_mode